lib/blocks.php

```diff
@@ -295,36 +295,48 @@ function gutenberg_experimental_apply_classnames_and_styles( $block_content, $bl
  * Add CSS classes and inline styles for colors to the incoming attributes array.
  * This will be applied to the block markup in the front-end.
  *
- * @param  array $attributes comprehensive list of attributes to be applied.
- * @param  array $block_attributes block attributes.
- * @param  array $supports style features the block attributes.
+ * @param  array  $attributes comprehensive list of attributes to be applied.
+ * @param  array  $block_attributes block attributes.
+ * @param  array  $supports style features the block attributes.
+ * @param  object $block_type registered block type class.
  * @return array Colors CSS classes and inline styles.
  */
-function gutenberg_experimental_build_css_colors( $attributes, $block_attributes, $supports ) {
+function gutenberg_experimental_build_css_colors( $attributes, $block_attributes, $supports, $block_type ) {
```

Comment:
This function is getting really long, and has a lot of conditionals. Not in this PR, but I think it would be good to consider ways to simplify and break out different parts into separate functions. With the number of conditionals it's becoming harder to follow.